src/components/inputs/FlySelect/FlySelect.tsx

```diff
@@ -175,6 +189,63 @@ export default class FlySelect extends React.Component<IProps, IState> {
 		};
 	}
 
+	onContainerKeyDown (event: any) {
+		let open = this.state.open;
+		let focusedIndex = this.state.focusedIndex;
+		switch (event.key) {
+			case 'Enter':
+				open = true;
+				break;
+			case 'ArrowUp':
+				open = true;
+				if (focusedIndex > 0) {
+					focusedIndex--;
+				}
+				break;
+			case 'ArrowDown':
+				open = true;
+				if (focusedIndex < Object.keys(this.state.optionsFormatted).length - 1){
+					focusedIndex++;
+				}
+				break;
+			case 'Tab':
+				open = false;
+				focusedIndex = 0;
+				break;
+		}
+
+		this.setState({
+			focusedIndex,
+			open,
+		}, () => {
+			if (this.state.open){
+				this.__optionsRef.current.children[this.state.focusedIndex] &&
+				this.__optionsRef.current.children[this.state.focusedIndex].focus();
+			}
+		});
+	}
+
+	onOptionKeyDown = (e: any, value: any) => {
+		if (e.key === 'Enter'){
+			this.selectOption(e, value);
+		}
+		this.__containerRef.current.focus();
+	}
+
+	onClickOutside (event: any) {
+		try {
+			const domNode = ReactDOM.findDOMNode(this.__containerRef.current);
+
+			if (!domNode || !domNode.contains(event.target)) {
+				this.setState({
+					focus: false,
+					open: false,
+				});
+			}
+		}
+		catch (error) {}
+	}
+
```

Comment:
So awesome including this!!!

Reply:
Curious, what issues did you run into with `onBlur` that necessitated `onClickOutside`?

Reply:
So, I originally added that because the onBlur handler was called on the container div. So when I tried to add the open/close functionality, since the focus shifted to whatever option was navigated to, the blur handler was called so the options would immediately disappear.

Reply:
Aha, makes sense!

Maybe something like https://stackoverflow.com/a/60094794 would simplify this?

Reply:
Oh wild, I hadn't seen a solution like that before

Reply:
That is a good option, but I don't think it's necessary. I removed the onClickOutside code, and moved the blur handler from the select container div to the options container div. That mitigated the issue with just 1 line of code added to the blur handler toggling the open/close state.